java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/OsmAndMapsService.java

```diff
@@ -793,6 +803,9 @@ public synchronized List<RouteSegmentResult> routing(String routeMode, Map<Strin
 			usedMapList = getReaders(list);
 			
 			RoutingContext ctx = prepareRouterContext(routeMode, points, router, rsc, avoidRoadsIds, usedMapList);
+			if (hhOnlyForce) {
```

Comment:
Better to calculate the force switch here - router.USE_ONLY_HH_ROUTING, without passing extra params

Reply:
In this method, router is not defined before method was called, so we need hhOnlyForce param to set the USE_ONLY_HH_ROUTING flag.

Reply:
RoutePlannerFrontEnd router = new RoutePlannerFrontEnd();

`router` is a dynamically created class in this method, so using hhOnlyForce as distinct parameter to change variable in just created class instance is Okay.

It's basically not so good idea to have USE_ONLY_HH_ROUTING (etc!) defined as static, because OsmAndServer is a **potentially multi-threaded** App and should not have any "global" static variables as settings which might be defined independendly...